src/admin/sections/CreateBallotForm.tsx

```diff
@@ -0,0 +1,129 @@
+import { FormEvent, useState } from "react";
+import { VoteOption } from "../../models";
+import { addBallot } from "../../network/ballotApi";
+import { Jwt } from "../../network/jwt";
+import VoteOptionView from "../components/VoteOption";
+import styles from "./CreateBallotForm.module.scss";
+
+interface CreateBallotProps {
+  jwt: Jwt;
+  onSubmit(): void;
+  onCancel(): void;
+}
+
+export function CreateBallotForm({
+  jwt,
+  onSubmit,
+  onCancel,
+}: CreateBallotProps) {
+  const [error, setError] = useState<string | null>(null);
+  const [showErrors, setShowErrors] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+
+  const [running, setRunning] = useState(false);
+  const [question, setQuestion] = useState("");
+  const [voteOptions, setVoteOptions] = useState<VoteOption[]>([]);
+
+  function submit(e: FormEvent) {
+    e.preventDefault();
+
+    if (question === "") {
+      setError("Error: Question missing");
+      setShowErrors(false);
+    } else if (voteOptions.length < 2) {
+      setError("Error: You need at least 2 vote options");
+      setShowErrors(false);
+    } else if (
+      voteOptions.find((vo) => vo.identifier === "" || vo.label === "") != null
+    ) {
+      setError("Error: Some values are missing");
+      setShowErrors(true);
+    } else {
+      setShowErrors(true);
+      setDisabled(true);
```

Comment:
`disabled` is never getting set to false and is basically useless right now.

Reply:
It is set to `false` initially. It makes sure you can't create 2 ballots by clicking the "Submit" button twice: When it is first clicked, the button is disabled, and when the request is finished, the entire form is hidden.